pkg/kvstore/allocator/allocator.go

```diff
@@ -590,18 +590,33 @@ func (a *Allocator) Get(ctx context.Context, key AllocatorKey) (idpool.ID, error
 	return a.GetNoCache(ctx, key)
 }
 
+func prefixMatchesKey(prefix, key string) bool {
+	// cilium/state/identities/v1/value/label;foo;bar;/172.0.124.60
+	//
+	// Presumably node IPs could also be IPv6
+	k := strings.TrimRight(key, "/0123456789abcdef:")
```

Comment:
This assumes that the key version of the labels list ends with a semicolon. This appears to be the case based on real entries dumped from a test Cilium cluster.